scipy/stats/unuran/unuran_wrapper.pyi

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+from typing import TYPE_CHECKING, Union, Any, Callable, Tuple, Optional, List
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+SeedType = Union[np.random.RandomState, np.random.Generator, int,
+                 npt.ArrayLike]
+
+
+__all__: List[str]
+
+
+class Method:
+    ...
+
+
+class TransformedDensityRejection(Method):
+    def __init__(self,
+                 pdf: Callable, dpdf: Callable,
```

Comment:
```suggestion
                 pdf: Callable[..., Any], dpdf: Callable[..., Any],
```
Style nit: when using a parametrized type without any explicit parameters it's considered best practice to parametrize them with `Any`.

Reply:
Based on the signature described in the docstring, you could actually use [`__call__`-based protocols](https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols) to provide a better description for `pdf` and `dpdf`:
``` python
from typing_extensions import Protocol

class _PDFFunc1(Protocol):
    def __call__(self, __x: float) -> float: ...

class _PDFFunc2(Protocol):
    def __call__(self, __x: float, *args: Any) -> float: ...

# `_PDFFunc2` is unfortunately not a supertype of `_PDFFunc1`,
# so we have to use a union here
_PDFFunc = Union[_PDFFunc1, _PDFFunc2]
```

Reply:
I have updated this now. After a discussion with @chrisb83, the API has been changed to accept a single `dist` object with all the required methods packed in it. I have tried to write protocols for it. Can you check them and let me know if I am doing something wrong?

Reply:
Looks pretty good overall, though I would suggest one change:
``` python
from typing import Callable
from typing_extensions import Protocol

class TDRDist(Protocol):
    @property
    def pdf(self) -> Callable[..., float]: ...
    @property
    def dpdf(self) -> Callable[..., float]: ...
```
I never realized this, but `*args` and `**kwargs` is currently a bit of disaster in conjunction with protocols.
Namely, the argument types represent a lower bound (because contravariance) to what concrete implementation 
must implement. So adding `*args` means that the user supplied function _must_ also have `*args` in its signature.

Annoyingly enough the only way around this issue seems to be the use of `Callable`, 
even though it's not as specific as one would ideally like.

Reply:
Thanks for taking a look!

> I never realized this, but `*args` and `**kwargs` is currently a bit of disaster in conjunction with protocols.
> Namely, the argument types represent a lower bound (because contravariance) to what concrete implementation
> must implement. So adding `*args` means that the user supplied function _must_ also have `*args` in its signature.

I was worried about this. I will change it to use `property` as you suggest.

Reply:
Done.

Reply:
It's also possible to do it without `@property`, but then you're more likely to run into issues later,
_i.e._ when you try to a subtype the mutable attribute with an immutable `__call__` method.

``` python
class TDRDist(Protocol):
    pdf: Callable[..., float]
```